qt5-static package is broken since 5.15.14-1

### Description / Steps to reproduce the issue

Try to build any cmake-using project which uses it.

### Expected behavior

cmake finds qt5-static and all its dependencies successfully.

### Actual behavior

It crashes when looking for libz with this:
```
CMake Error at C:/msys64/mingw32/qt5-static/lib/cmake/Qt5Core/Qt5CoreConfig.cmake:91 (message):
  Library not found: :libz
Call Stack (most recent call first):
  C:/msys64/mingw32/qt5-static/lib/cmake/Qt5Core/Qt5CoreConfig.cmake:272 (_qt5_Core_process_prl_file)
  C:/msys64/mingw32/qt5-static/lib/cmake/Qt5/Qt5Config.cmake:28 (find_package)
  src/qt/CMakeLists.txt:44 (find_package)


-- Configuring incomplete, errors occurred!
[!] CMake failed with status [1]
[Pipeline] writeFile
[Pipeline] }
[Pipeline] // dir
[Pipeline] }
[Pipeline] // dir
[Pipeline] catchError
[Pipeline] {
[Pipeline] }
Also:   org.jenkinsci.plugins.workflow.actions.ErrorAction$ErrorId: 27f24933-88ed-4e28-951d-97c00e10e173
java.lang.ArithmeticException: Division by zero
```

### Verification

- [X] I have verified that my MSYS2 is up-to-date before submitting the report (see https://www.msys2.org/docs/updating/)

### Windows Version

MINGW64_NT-10.0-22631

### MINGW environments affected

- [X] MINGW64
- [X] MINGW32
- [ ] UCRT64
- [ ] CLANG64
- [ ] CLANG32
- [ ] CLANGARM64

### Are you willing to submit a PR?

Yes.

Doesn't work for me:
```
CMake Error at C:/msys64/mingw64/qt5-static/lib/cmake/Qt5Core/Qt5CoreConfig.cmake:91 (message):
  Library not found: :libz.a
Call Stack (most recent call first):
  C:/msys64/mingw64/qt5-static/lib/cmake/Qt5Core/Qt5CoreConfig.cmake:272 (_qt5_Core_process_prl_file)
  C:/msys64/mingw64/qt5-static/lib/cmake/Qt5/Qt5Config.cmake:28 (find_package)
  src/frontend/qt_sdl/CMakeLists.txt:73 (find_package)


-- Configuring incomplete, errors occurred!
```

Only replacing with `-lz` and `-lzstd` works.

--- a/mingw-w64-qt5-static/PKGBUILD
+++ b/mingw-w64-qt5-static/PKGBUILD
@@ -6,19 +6,19 @@ _realname=qt5-static
 pkgbase="mingw-w64-${_realname}"
 pkgname="${MINGW_PACKAGE_PREFIX}-${_realname}"
 pkgver=5.15.14
-pkgrel=1
+pkgrel=2
 pkgdesc="A cross-platform application and UI framework (static) (mingw-w64)"
 arch=('any')
 mingw_arch=('mingw32' 'mingw64' 'ucrt64' 'clang64' 'clang32' 'clangarm64')
 url='https://www.qt.io/'
 msys2_references=(
   'archlinux: qt5-base'
+  'aur: mingw-w64-qt5-base-static'
   "cpe: cpe:/a:qt:qt"
 )
 msys2_repository_url='https://github.com/qt/qt5'
 license=('spdx:GPL-2.0-only WITH Qt-GPL-exception-1.0 AND GPL-3.0-or-later AND LGPL-2.1-or-later AND LGPL-3.0-only AND GFDL-1.3-or-later')
-depends=("${MINGW_PACKAGE_PREFIX}-gcc-libs"
-         "${MINGW_PACKAGE_PREFIX}-vulkan"
+depends=("${MINGW_PACKAGE_PREFIX}-vulkan"
          "${MINGW_PACKAGE_PREFIX}-z3")
 makedepends=("${MINGW_PACKAGE_PREFIX}-cc"
              "${MINGW_PACKAGE_PREFIX}-fxc2"
@@ -365,6 +365,10 @@ package() {
                  -e "s|-L\$\$\[QT_INSTALL_LIBS\] ||g" \
                  -e "s|-L\$\$\[QT_INSTALL_LIBS\];||g" \
                  -e "s|.a | |g" -e "s|.a;|;|g" \
+                 -e "s|-l\:libz\;|-llibz.a\;|g" \
+                 -e "s|-l\:libzstd\;|-llibzstd.a\;|g" \
+                 -e "s|-l\:libz |-llibz.a |g" \
+                 -e "s|-l\:libzstd |-llibzstd.a |g" \
                  -e "s|lqt_clipper.a|lqt_clipper|g" {} \;
 
   find "${pkgdir}${MINGW_PREFIX}/${_realname}/lib" -type f -name '*.prl' \
@@ -374,5 +378,9 @@ package() {
                  -e "s|-L\$\$\[QT_INSTALL_LIBS\] ||g" \
                  -e "s|-L\$\$\[QT_INSTALL_LIBS\];||g" \
                  -e "s|.a | |g" -e "s|.a;|;|g" \
+                 -e "s|-l\:libz\;|-llibz.a\;|g" \
+                 -e "s|-l\:libzstd\;|-llibzstd.a\;|g" \
+                 -e "s|-l\:libz |-llibz.a |g" \
+                 -e "s|-l\:libzstd |-llibzstd.a |g" \
                  -e "s|lqt_clipper.a|lqt_clipper|g" {} \;
 }

